pdm/models/repositories.py

```diff
@@ -112,6 +112,13 @@ def find_candidates(
             reverse=True,
         )
 
+        if not sorted_cans:
+            self.environment.project.core.ui.echo(
```

Comment:
Should these messages be written into the log file when verbose is off?

Reply:
Good idea. What would you prefer? I'm fine with both.

Reply:
`logger` is preferred for this case, it can be imported from `termui` or got with the exact same name.